onnx/onnx.proto

```diff
@@ -52,17 +52,52 @@ enum Version {
   // version that the  graph is generated from. This helps us set up version
   // control. We should use version as
   //     xx(major) - xx(minor) - xxxx(bugfix)
-  // and we are starting with 00000001.
-  IR_VERSION = 00000001;
+  // and we are starting with 00000001, which was the
+  //  version we published on Oct 10, 2017.
+  IR_VERSION_2017_10_10 = 0x00000001;
+
+  // IR_VERSION 1.0.1 (this proto file) contains the following IR format changes
+  // - Added type discriminator to AttributeProto to support proto3 users
+
+  IR_VERSION = 0x00010001;
 }
 
+
+
 // A named attribute containing either singular float, integer, string
 // and tensor values, or repeated float, integer, string and tensor values.
 // An AttributeProto MUST contain the name field, and *only one* of the
 // following content fields, effectively enforcing a C/C++ union equivalent.
 message AttributeProto {
+
+  // Note: this enum is structurally identical to the OpSchema::AttrType
+  // enum defined in schema.h.  If you rev one, you likely need to rev the other.
+  enum AttributeType {
```

Comment:
So, in #78 I mentioned that we might want to stop using protobuf nesting. Does this apply to enums as well?

Reply:
For message types, we should absolutely stop nesting.

For enums, I would argue that we should nest unless we want to inject the individual enum identifiers into the global/package namespace.

For the example above, when I had it at the top level, the identifier STRING shows up as a onnx.STRING in python.  It's actually worse - if you look at AttributeType and the DataType enum from TensorProto, we would have two conflicting identifiers (that would at best accidentally have the same value).

I think our style should be NO nesting of messages and ALWAYS nesting of enums.  I'd leave IR_VERSION as is just for historical reasons.

Objections?

Reply:
Nope, you've convinced me. Nested enums it is!